UnityProject/Assets/Scripts/Core/SecureStuff/SecureMapsSaver.cs

```diff
@@ -68,81 +87,557 @@ public virtual void AddID(string ToAdd)
 		public virtual void Serialise()
 		{
 			if (ReferencingIDs == null) return;
+			bool First = true;
 			foreach (var ID in ReferencingIDs)
 			{
-				Data = Data + "," + ID;
+				if (First)
+				{
+					Data = Data + ID;
+				}
+				else
+				{
+					Data = Data + "," + ID;
+				}
+
+				First = false;
 			}
 		}
 
 		public string Name;
 		public string Data;
+		public bool? IsPrefabID;
 	}
 
 	public static class SecureMapsSaver
 	{
-		public static void RecursiveSearchData(HashSet<Component> OnGameObjectComponents,
-			HashSet<GameObject> AllGameObjectOnObject,
-			IPopulateIDRelation IPopulateIDRelation,
-			HashSet<FieldData> FieldDatas,
-			string Prefix,
-			object PrefabInstance,
-			object SpawnedInstance,
-			bool UseInstance = false)
+		private static void ListHandle(string RootID, Component Root, FieldInfo Field, object Object, FieldData ModField, int Index,
```

Comment:
fix codacy errors

Reply:
this function is way too large, split into smaller functions that have self-explainable names